consensus/src/config.rs

```diff
@@ -100,25 +96,47 @@ pub fn exponential_slowdown(
     Duration::from_millis(delay)
 }
 
-/// A default configuration of what the creators of this package see as optimal parameters.
-pub fn default_config(n_members: NodeCount, node_ix: NodeIndex, session_id: SessionId) -> Config {
-    Config {
+/// Creates a [`Config`], allowing the user to omit specifying `max_round` and `delay_config`
+/// in which case they will be set do default values, suggested by the creators of this package.
+/// `time_to_reach_max_round` is a lower bound on the time needed to reach the maximum round
+/// expected by the user and is only used for verification.
+pub fn create_config(
+    n_members: NodeCount,
+    node_ix: NodeIndex,
+    session_id: SessionId,
+    max_round: Option<Round>,
+    delay_config: Option<DelayConfig>,
```

Comment:
This isn't a great interface, if you want to be able to construct something with default values then write a separate constructor and document what guarantees it gives (in particular what the approximate `time_to_reach_max_round` is – it makes little sense to pass this as an argument if you are using the default). Then this constructor can always require these arguments.

Reply:
Makes sense. I propose 2 functions: `create_config` which has an argument for each field of `Config` and `create_config_with_default_delays` which skips the `delay_config` argument. Each of these would take `time_to_reach_max_round` and perform the check.

Reply:
I think there are 3 types of configs that we will ever need to create:
1. Config for clueless users: this needs to take `n_members`, `node_ix` and `session_id`. The rest should be some sensible defaults. We can also include a check that `time_to_reach_max_round=7days` works (so that we never mess up the defaults inside of AlephBFT).  
2. Config for us, `aleph_node`: we should be able (and actually should be forced) to customize every single field, this should also require us to specify `time_to_reach_max_round` explicitly. We really want to avoid the situation like before that we use the default config, but then change a single field which then gets us into trouble. 
3. Test configs.

I guess the 2 constructors that you propose should cover well all 3 cases.

Reply:
I would just suggest `default_config` as a less unwieldy name for the second one.